### DISCUSSION TITLE
Unable to pair Sonoff SNZB-02

### DISCUSSION THREAD

**Discussion by matbech** [Koenkk/zigbee2mqtt#8256] [Devices]
I'm unable to pair Sonoff SNZB-02 with Zigbee2MQTT.

What I have tried:
- Restarted system where the coordinator is connected to
- Hold the button on the SNZB-02 device until the LED was flashing 3 times
- Monitored the Zigbee2MQTT log
- No message from the device shows up in the Zigbee2MQTT log
- This is a new device with a new battery

**Device**
SONOFF SNZB-02 ZIGBEE TEMPERATURE & HUMIDITY SENSOR
https://sonoff.tech/product/smart-home-security/snzb-02/

**configuration.yaml**
```
homeassistant: true
permit_join: true
mqtt:
  base_topic: zigbee2mqtt
  server: 'mqtt://localhost'
serial:
  port: /dev/ttyACM0
  disable_led: true
```

**log.txt**
```
info  2021-08-04 15:24:26: Starting Zigbee2MQTT version 1.21.0 (commit #70891eec)
info  2021-08-04 15:24:26: Starting zigbee-herdsman (0.13.124)
info  2021-08-04 15:24:30: zigbee-herdsman started (resumed)
info  2021-08-04 15:24:30: Coordinator firmware version: '{"meta":{"maintrel":3,"majorrel":2,"minorrel":6,"product":0,"revision":20190608,"transportrev":2},"type":"zStack12"}'
info  2021-08-04 15:24:30: Currently 7 devices are joined:
...
warn  2021-08-04 15:24:30: Allowing new devices to join.
warn  2021-08-04 15:24:30: Set `permit_join` to `false` once you joined all devices.
info  2021-08-04 15:24:30: Zigbee: allowing new devices to join.
info  2021-08-04 15:24:30: Connecting to MQTT server at mqtt://localhost
info  2021-08-04 15:24:30: Connected to MQTT server
```

Coordinator: CC2531
Firmware: zStack revision: 20190608
and:
Firmware: zStack revision: 20201127

Do I need the zStack 3.x firmware for Zigbee 3.0 support?

**Comment by sjorge** [comment]
I got a bunch of those and they are a bit annoying to pair at times. Make sure if a previous failed attempt left a entry in the web interface to force remove it before retrying.

When it started the interview and if it seems stuck, I sometimes remove and insert the battery again and it continues.

**Reply by jbramwell** [reply]
I had this issue as well with a set of SNZB-02 sensors I was adding to my Zigbee2MQTT network. Based on some experimentation and research, I came up with a repeatable process that has been working reliably for me (YMMV). I documented the process [in this blog post](https://tronics.life/connecting-the-sonoff-snzb-02-and-cc2531-zigbee-coordinator/) in case anyone else has this issue down the road.

**Reply by chojnicki** [reply]
@jbramwell Thaaaank you for that tip with shortcircut of battery slot. I lost like a two hours trying to pair multiple new sensors, and this finally helped...

**Reply by gaggio** [reply]
@jbramwell I'd like to try the trick, which seems successful, but the blog post doesn't seem to be accessible anymore. Do you have an alternative link or can explain here?
many thanks

**Reply by jbramwell** [reply]
@gaggio - yeah, my blog URL changed recently. [Here's the new link](https://bitsandbytes.life/2023/02/connecting-the-sonoff-snzb-02-and-cc2531-zigbee-coordinator) to that same blog post. I hope it helps!

**Reply by paperdigits** [reply]
@jbramwell the photos in the post still point to your old URL.

**Reply by jbramwell** [reply]
@gaggio - DOH! My bad 😁 I've re-enabled the old site until I get the images moved over properly (thanks for the heads up! Not sure why I didn't notice this earlier?). You should be able to view the entire post now.

**Reply by paperdigits** [reply]
@jbramwell thank you!

**Comment by matbech** [comment]
Pairing this sensor was a chore. After several attempts (including removing the battery etc), it finally paired.

**Comment by dylan09** [comment]
Same here. Pairing Sonoff Devices is a little bit of a nightmare. Often I also ended up in removing and inserting battery to get them running.
When it seems to be impossible to pair a device sometimes it helps to change the battery and use a fresh one.

In the future I will stay with the Aquara temperature sensors. They work like a charm. Pairing is fast and reliable.

**Reply by matbech** [reply]
> In the future I will stay with the Aquara temperature sensors. They work like a charm. Pairing is fast and reliable.

My Aquara sensors don't want to go through my IKEA Tradfri repeaters (apparently a known issue). That is the reason I wanted to see if it works with the Sonoff sensors.

**Comment by sjorge** [comment]
Yeah they are very temperamental, if the initial pair didn't 100% complete, it's usually wipe the leftovers, remove battery, insert battery, repeat until working.

Once they were I find them to be pretty solid, even switching routers when needed though. And the reporting is fully configurable, so I put up with it.

**Comment by cyijun** [comment]
I had the same problem until using the zstack 3.x.x firmware, but my cc2530 coordinator became unstable. That's the reason why I design a cc2652 hardware!🤣 Sonoff sucks

**Comment by matbech** [comment]
Exactly the same experience here, once I switched to cc2562 they paired on the first attempt and they even got rerouted through the IKEA Tradfri repeaters. Something the xiaomi/aqara devices don't do. Maybe the incompatibility with zstack 1.x should be noted somewhere to save someone else a lot of frustration.

**Reply by cyijun** [reply]
Yes, you can commit this to the supported-devices page of zigbee2mqtt.io.

**Comment by paperdigits** [comment]
I have 3 switches and 2 temperature sensors from Sonoff that absolutely refuse to pair anymore. I've tried many batteries and just about everything I can find, including the things on this thread. 

I don't see anything about them when zigbee-herdsman is set to debug mode. It seems they don't even try to join the network. 

I recently switched the channel of my zigbee stick. I forcibly removed everything from the network, change the channel to 25, then repaired everything. Everything repaired, but these switches and temp sensors fell off the network. I was able to repair them a few times, but now they've just quit. 

I have no idea what to do anymore.

**Comment by silviuchingaru** [comment]
@paperdigits try channel 24. See why in [ZigBee and Wi-Fi Coexistence](https://www.metageek.com/training/resources/zigbee-wifi-coexistence/) !

**Comment by paperdigits** [comment]
Thanks. I'm getting a bunch of new devices, so I'll going to start over in. a few months.

These buttons and sensors were previously pared on ch. 25, but fell off the network and will not rejoin.

When I start over I will also try some fresh, brand name batteries as well.

Thanks for the suggestion!

**Comment by chojnicki** [comment]
I lost many hours because of them. If they do not pair - reset them like suggested before (battery socket) and what finally worked for me - connect them by coordinator and not repeater (force i Z2M and put close to coordinator). I have more than 100 devices, but only them I had to connect this way.

**Comment by silviuchingaru** [comment]
Yes, I had problems with repeaters also, at least with BlitzWolf SHP-13 devices as repeters. If i try to join over repeters they don't work but, if I join next to coordonator, they work well over repeters also. I use Tasmota ZigBee, by the way. If you go fresh, maybe is a good ideea to change also the PanID with the channel.

**Comment by paperdigits** [comment]
I ended up going back to zigbee channel 11 and move my wifi to a different, non-overlapping channel. The buttons and other battery things are connecting fine now. Thanks!

**Reply by WenderGalan** [reply]
I had changed my router and I didn't notice that it was causing a problem, now about 5 days later I noticed that several ZigBee devices were messed up, with no bad communications to connect and everything, I saw your comment and remembered the router, the router was on channel 1 (2.4 Ghz) and zigbee on channel 11, I changed the router to channel 14 and oddly enough all the problems disappeared, the zigbee network was apparently very stable.

**Comment by delangen** [comment]
For me pairing SNZB-02's failed until I tried do pair close to the coordinator.

**Reply by Retr0Joe** [reply]
I have the same issue. I cannot pair any SNZB-02 devices over (IKEA tradfri outlets) routers. Only when i put them near to the coordinator, it is possible to pair them. I use Skyconnect with firmware v4.3.1.0 via Zigbee and OpenThread multiprotocol add-on and ezsp configuration at zigbee2mqtt. Sometimes, they disconnect randomly for no reason. After repairing, they work again.

**Reply by KrisPrzybyla** [reply]
Thanks a lot, @delangen! I finally managed to pair the SNZB-02.

**Comment by dave-sausages** [comment]
I know this is an old thread, but I just solved this issue using a method not described here. In the end I got my 5 SNZB-02 devices to pair by powering down every coordinator light bulb and repeater and just having the coordinator on my home assistant pi running. I don't know if I needed to power down ALL of the devices I did, but it worked instantly when I did.
